app/javascript/packs/leafletMap.js

```diff
@@ -64,14 +64,42 @@ export async function setupMap() {
   map.on("click", onClick);
 }
 
+function parsePosition(position){
+  return "?long="+position.lng+"&lat="+position.lat
+}
+
+function pinPopupContent(position) {
+  return "<ul>"+
+  "<li><a href=\"/rooms/new"+parsePosition(position)+"\">Add Room</a></li>"+
+  "<li><a href=\"/buildings/new"+parsePosition(position)+"\">Add Building</a></li>"+
+  "<li><a href=\"/locations/new"+parsePosition(position)+"\">Add Location</a></li>"+
+  "<li><a id=\"deletepin\" \" href=\"javascript:void(0);\" >Delete Pin</a></li>"+
+  "</ul>";
+}
```

Comment:
Javascripts backtick-strings (aka [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)) make this a bit nicer:
```javascript
  const params = parsePosition(position);
  return `
    <ul>
      <li><a href="/rooms/new"${params}">Add Room</a></li>
      <li><a href="/buildings/new"${params}">Add Building</a></li>
      <li><a href="/locations/new"${params}">Add Location</a></li>
      <li><a id="deletepin" href="javascript:void(0);">Delete Pin</a></li>
    </ul>
  `;
```